lib/fluent/plugin/buffer/chunk.rb

```diff
@@ -86,9 +86,16 @@ def modified_at
         # data is array of formatted record string
         def append(data, **kwargs)
           raise ArgumentError, '`compress: gzip` can be used for Compressable module' if kwargs[:compress] == :gzip
-          adding = ''.b
-          data.each do |d|
-            adding << d.b
+          begin
+            adding = data.join.force_encoding(Encoding::ASCII_8BIT)
+          rescue
+            # Fallback
+            # Array#join throws an exception if data contains strings with a different encoding.
+            # Although such cases may be rare, it should be considered as a safety precaution.
+            adding = ''.force_encoding(Encoding::ASCII_8BIT)
+            data.each do |d|
+              adding << d.b
```

Comment:
> adding << d.b

Is the use of `String#b` only here for performance?

Reply:
`String#b` will duplicate the `self` and change encoding of duplicated to `ASCII-8BIT`
https://docs.ruby-lang.org/en/3.4/String.html#method-i-b

Maybe, it didn't want to directly change the encoding of the original string.

Since this method results in many duplicate strings, 
this patch uses `Array#join` due to suppress allocate the Object.

Reply:
> Maybe, it didn't want to directly change the encoding of the original string.

I see. Thanks!